### ISSUE TITLE
Sample from the model

### ISSUE THREAD

**Issue by chief7** [keithito/tacotron#2]
Hi! First of all: Thank you very much for the good work. Very nice an clean code!

Anyway: I've trained my model for ~100k global steps so far and the audio outputs generated during training are very nice and clean voices. But if I try to sample something using the provided demo server, all I get is gibberish.

Before I give the model some more days to train: Did you sample you're examples from the model using the server (or the any other method) or did you just take what the training process puts into the log directory?

Any hint would be much appreciated. Thanks again!

**Comment by keithito** [comment]
Hi! Sorry to hear you're getting gibberish. Which dataset are you using?

The examples were generated using eval.py, which takes the same --checkpoint argument as the demo server (the code is very similar).

The eval sentences are from the New York Times from last Saturday and from Google's Tacotron paper, so I don't think there's much overlap between them and the training data.

**Comment by chief7** [comment]
Thanks for the quick answer!
I'm using a custom dataset. I'll let the ljspeech set run for some hours and report back as soon as possible. Let's see if that changes anything...

**Comment by chief7** [comment]
Just one more thing: do you remember after how many steps the model started 
to generate recognizable words?


Am 14. Juli 2017 9:24:10 nachm. schrieb Keith Ito <notifications@github.com>:

**Comment by keithito** [comment]
It started to become intelligible around 20,000 global steps.

**Comment by sonach** [comment]
Very good job! I am trying it too.

**Comment by sonach** [comment]
I have completely reproduce the result. Although the results are not as good as tacotron author's samples, this project is a very good baseline.

**Comment by DarkDefender** [comment]
@keithito Any plans to train further? I'm asking because you removed that statement from the readme recently...

 I don't know if it is comparable, but if I understand correctly then the paper talk about training to 2M steps:

> with learning rate decay, which starts from 0.001 and is reduced to 0.0005, 0.0003, and 0.0001 after 500K, 1M and 2M global steps, respectively.

If your loss is still going down, then perhaps it will improve enough from ~1M to 2M that it is worth it?

**Comment by keithito** [comment]
Yeah, I'm planning on going up to 2M. I needed to use my GPU for some unrelated tasks for a few days, but I'll get back to training today probably :-)

**Comment by DarkDefender** [comment]
Ok, thanks for the update! :)

**Comment by GuangChen2016** [comment]
@keithito Hello, very nice jobs. I have tried to use your repo to train a Tractron model in my own data set. But after 4600 steps, I got an error as below. I doubt it may be caused by the Loss Exploding, and I decreased the learning rate to a small one, like 0.0001, 0.00001. But it didn't work. Could you provide some suggestions for me? (But when I tried on the ljspeech data set, it worked fine)
[2017-08-03 03:01:15.242]  Step 4599    [1.134 sec/step, loss=0.11571, avg_loss=0.11422]
[2017-08-03 03:01:16.504]  Step 4600    [1.135 sec/step, loss=0.11646, avg_loss=0.11422]
[2017-08-03 03:01:16.504]  Writing summary at step: 4600
[2017-08-03 03:01:17.955]  Exiting due to exception: Nan in summary histogram for: model/stats/gradient_norm
Thank you very much.

**Comment by matanster** [comment]
@sonach care to share what have been the characteristics of your dataset you used for training, and how long did the training take on your hardware, for that dataset?

**Comment by keithito** [comment]
@matanster I trained on the [LJ Speech dataset](https://keithito.com/LJ-Speech-Dataset/). See that page for more details on the dataset. I trained for about 900K steps with batch size 32, which took about 12 days on a GTX 1080Ti. But there wasn't much noticeable difference after 250K steps / 3 days.

**Comment by matanster** [comment]
thanks @keithito that's very interesting, as the voice quality I see in that dataset (judging from the samples at https://keithito.com/LJ-Speech-Dataset/) is quite good by my human judgement, and the overall duration is the same as google's internal dataset used in their original research. It remains somewhat elusive then, how it worked much better for them, if it really did.
